src/main/java/org/hydev/mcpm/client/commands/entries/MirrorController.java

```diff
@@ -0,0 +1,94 @@
+package org.hydev.mcpm.client.commands.entries;
+
+import org.hydev.mcpm.client.database.mirrors.MirrorSelectBoundary;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.hydev.mcpm.utils.FormatUtils.tabulate;
+
+/**
+ * Controller for the mirror selection command
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-22
+ */
+public record MirrorController(MirrorSelectBoundary boundary)
+{
+    /**
+     * Format ping with colors
+     *
+     * @param ping Ping in milliseconds
+     * @return Ping with colors
+     */
+    private static String formatPing(long ping)
+    {
+        var color = ping <= 100 ? "&a" : ping <= 200 ? "&e" : "&c";
```

Comment:
Embedded ternary is sane but hard to read. If this is going to be its own method I would expand this and `formatSpeed` out into at least one if statement.